src/main/utils.ts

```diff
@@ -264,3 +267,57 @@ export const installApp = async ({
     });
   }
 };
+
+export async function factoryResetUtility({
+  launcherFileSystem,
+  windowInfoMap,
+  privilegedLauncherWindows,
+  holochainManagers,
+  lairHandle,
+  app,
+}: {
+  launcherFileSystem: LauncherFileSystem;
+  windowInfoMap: WindowInfoRecord;
+  privilegedLauncherWindows?: Record<Screen, BrowserWindow>;
+  holochainManagers: Record<string, HolochainManager>;
+  lairHandle?: ChildProcessWithoutNullStreams;
+  app: Electron.App;
+}) {
+  if (!launcherFileSystem) {
+    throw new Error('LauncherFilesystem is undefined. Aborting Factory Reset.');
+  }
+
+  // Helper function to close windows
+  const closeWindows = (windows: WindowInfoRecord | Record<string, BrowserWindow>) => {
+    Object.values(windows).forEach((window) => window.close());
```

Comment:
I don't think this will work as the values of `WindowInfoRecord` do not have a `close()` method, they are of type `WindowInfo` not `BrowserWindow`, the browser window is nested inside it.

Reply:
I don't see much value in defining a dedicated function for this anyway, especially since it's not being used anywhere else. It think makes the code less readable actually.

Reply:
100% right, I've just pushed a fix